evaluation/logic_reasoning/run_infer.py

```diff
@@ -0,0 +1,395 @@
+        )
+        # Remove all existing handlers from logger
+        for handler in logger.handlers[:]:
+            logger.removeHandler(handler)
+        file_handler = logging.FileHandler(log_file)
+        file_handler.setFormatter(
+            logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+        )
+        logger.addHandler(file_handler)
+
+    if not skip_workspace_mount:
+        logger.info(f'Process-specific workspace mounted at {workspace_mount_path}')
+
+    # sandbox = DockerSSHBox()
+    logic_inference_path = os.path.join(workspace_mount_path, 'logic_inference.py')
+    if not os.path.exists(logic_inference_path):
+        shutil.copyfile('./evaluation/logic_reasoning/logic_inference.py', logic_inference_path)
+    logger.info(f'logic_inference.py copied to {workspace_mount_path}')
+
+    cache_dir = os.path.join(workspace_mount_path, '.cache_program')
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+            
+    # Prepare instruction
+    
+    with open('./evaluation/logic_reasoning/instruction.txt', 'r') as f:
+        instruction = f.read()
+
+    instance_logic_programs = instance['raw_logic_programs'][0].strip()
+    instruction = instruction.replace("[[dataset_name]]", metadata['dataset'])
+    instruction = instruction.replace("[[logic_programs]]", instance_logic_programs)
+    instruction = instruction.replace("[[logic_inference_path.py]]", logic_inference_path)
+    
+    # NOTE: You can actually set slightly different instruction for different agents
+    instruction += AGENT_CLS_TO_INST_SUFFIX.get(agent_class, '')
+
+    # Here's how you can run the agent (similar to the `main` function) and get the final task state
+    state: State = asyncio.run(
+        main(
+            instruction,
+            fake_user_response_fn=AGENT_CLS_TO_FAKE_USER_RESPONSE_FN.get(agent_class),
+            # sandbox=sandbox,
+        )
+    )
+    # ======= Attempt to evaluate the agent's edits =======
+    # If you are working on simplier benchmark that only evaluates the final model output (e.g., in a MessageAction)
+    # You can simply get the LAST `MessageAction` from the returned `state.history` and parse it for evaluation.
+
+    if state is None:
+        raise ValueError('State should not be None.')
+
+    final_message = ''
+    messages = []
+    for action, obs in reversed(state.history):
+        # if isinstance(act, MessageAction):
+        messages.append(obs.content)
+        # print("obs.content:", obs.content)
+        if str(obs.content) in ["'A'", "'B'", "'C'"]:
+            final_message = obs.content
+            break
+        
+    logger.info(
+        f'Final message: {final_message}'
+    )
+
+    test_result = get_test_result(
+        model_answer=final_message, ground_truth=instance['answer']
+    )
+
+    # Save the output
+    output = {
+        'instance_id': instance['id'],
```

Comment:
Maybe we can change this to either `id` or `instance_id`, currently when I run locally it cannot load and exclude the instances that have been processed.

Reply:
thanks, definitely true that the ```id``` and ```instance_id``` should be consistent. I will correct this typo.